plugins/elantp/elantp.quirk

```diff
@@ -6,6 +6,12 @@ GType = FuElantpHidDevice
 Plugin = elantp
 GType = FuElantpHidDevice
 
+# recovery device
+[DeviceInstanceId=I2C\NAME_Synopsys-DesignWare-I2C-adapter]
```

Comment:
Hmmm.... How is this not going to probe every single machine with an i2c adapter at address 0x15?

Reply:
I2C doesn't have any specification for what is a probe and what is a more destructive action.  So I'm worried specifically that sending `tx_buf` out on the i2c bus to that address might not be safe.  What if you happen to have a retimer on the i2c bus and that retimer happens to have a power cycle command corresponding to what's in `tx_buf` to probe an Elan touchpad?  You might sever your USB4 link whenever fwupd starts up and cause some data loss on a downstream device.

Reply:
> for what is a probe and what is a more destructive action

Valid. We get the "name" from the `cat /sys/class/i2c-adapter/i2c-*/name` e.g `nvkm-0000:01:00.0-bus-0002` but I agree in the `Synopsys-DesignWare-I2C-adapter` case this is super generic and likely to match on more hardware than just the Acer laptop I have here. I think restricting it to a DMI match might be enough in this case, right?

Reply:
>Synopsys-DesignWare-I2C-adapter case this is super generic and likely to match on more hardware than just the Acer laptop I have here

I get a `Synposys DesignWare I2C adapter` on my XPS 13.  
How about using `HIDRAW\*` like we did on Wacom?  Does that work properly?

> I think restricting it to a DMI match might be enough in this case, right?

I think lets fall back to that as last resort if there's no better way.

Reply:
> How about using HIDRAW\* like we did on Wacom? Does that work properly?

Oh I guess this wouldn't work.  The whole point is that this doesn't export as HID device, it's raw I2C device.

So yes, DMI match is what we need to move this to for anything with an Elan touchpad that can recover this way.

Reply:
>  The whole point is that this doesn't export as HID device, it's raw I2C device

Right. We do get the HID descriptor using command 0x0001 but I don't think that's at all standardized on all i2c devices.

Reply:
Yes, Can HID descriptor commands be read from the acpi table? > 
-> Device id : ELANXXXX:XX
-> i2c-address: 0x15
-> HID Descriptor Register Address: 0x0001

Reply:
I've added some DMI allow-listing. @superm1 can you verify it **does not** enumerate a i2c device on you XPS 13 please.

Reply:
As far as I can tell it's not doing anything.

```
$ sudo FWUPD_PROBE_VERBOSE=1 fwupdtool get-devices --plugins=elantp --daemon-verbose=FuEngine
18:43:47:0538 FuEngine             client certificate exists and working
18:43:47:0545 FuEngine             remote lvfs-testing not enabled, so skipping
18:43:47:0545 FuEngine             remote fwupd-tests not enabled, so skipping
18:43:47:0545 FuEngine             remote vendor not enabled, so skipping
18:43:47:0545 FuEngine             remote vendor-directory not enabled, so skipping
18:43:47:0545 FuEngine             468 components now in silo
18:43:47:0555 FuEngine             plugin logind is not enabled
18:43:47:0555 FuEngine             plugin ccgx is not enabled
18:43:47:0555 FuEngine             plugin csr is not enabled
18:43:47:0555 FuEngine             plugin vli is not enabled
18:43:47:0555 FuEngine             adding plugin /usr/local/lib/x86_64-linux-gnu/fwupd-plugins-3/libfu_plugin_elantp.so
18:43:47:0556 FuEngine             plugin linux_sleep is not enabled
18:43:47:0556 FuEngine             plugin jabra is not enabled
18:43:47:0556 FuEngine             plugin rts54hid is not enabled
18:43:47:0556 FuEngine             plugin altos is not enabled
18:43:47:0556 FuEngine             plugin bios is not enabled
18:43:47:0556 FuEngine             plugin ata is not enabled
18:43:47:0556 FuEngine             plugin nitrokey is not enabled
18:43:47:0556 FuEngine             plugin uefi is not enabled
18:43:47:0556 FuEngine             plugin steelseries is not enabled
18:43:47:0556 FuEngine             plugin coreboot is not enabled
18:43:47:0556 FuEngine             plugin synaptics_rmi is not enabled
18:43:47:0556 FuEngine             plugin superio is not enabled
18:43:47:0556 FuEngine             plugin thunderbolt is not enabled
18:43:47:0556 FuEngine             plugin synaptics_prometheus is not enabled
18:43:47:0556 FuEngine             plugin tpm is not enabled
18:43:47:0556 FuEngine             plugin cros_ec is not enabled
18:43:47:0556 FuEngine             plugin uefi_dbx is not enabled
18:43:47:0556 FuEngine             plugin dfu is not enabled
18:43:47:0556 FuEngine             plugin linux_swap is not enabled
18:43:47:0556 FuEngine             plugin synaptics_mst is not enabled
18:43:47:0556 FuEngine             plugin fresco_pd is not enabled
18:43:47:0556 FuEngine             plugin wacom_raw is not enabled
18:43:47:0556 FuEngine             plugin ep963x is not enabled
18:43:47:0556 FuEngine             plugin emmc is not enabled
18:43:47:0556 FuEngine             plugin msr is not enabled
18:43:47:0556 FuEngine             plugin fastboot is not enabled
18:43:47:0556 FuEngine             plugin solokey is not enabled
18:43:47:0556 FuEngine             plugin ebitdo is not enabled
18:43:47:0556 FuEngine             plugin rts54hub is not enabled
18:43:47:0556 FuEngine             plugin iommu is not enabled
18:43:47:0556 FuEngine             plugin nvme is not enabled
18:43:47:0556 FuEngine             plugin wacom_usb is not enabled
18:43:47:0556 FuEngine             plugin logitech_hidpp is not enabled
18:43:47:0556 FuEngine             plugin thelio_io is not enabled
18:43:47:0556 FuEngine             plugin pci_bcr is not enabled
18:43:47:0556 FuEngine             plugin linux_tainted is not enabled
18:43:47:0556 FuEngine             plugin redfish is not enabled
18:43:47:0556 FuEngine             plugin dell is not enabled
18:43:47:0556 FuEngine             plugin optionrom is not enabled
18:43:47:0556 FuEngine             plugin colorhug is not enabled
18:43:47:0556 FuEngine             plugin dell_esrt is not enabled
18:43:47:0556 FuEngine             plugin acpi_facp is not enabled
18:43:47:0556 FuEngine             plugin upower is not enabled
18:43:47:0556 FuEngine             plugin tpm_eventlog is not enabled
18:43:47:0556 FuEngine             plugin amt is not enabled
18:43:47:0556 FuEngine             plugin uefi_recovery is not enabled
18:43:47:0556 FuEngine             plugin pci_mei is not enabled
18:43:47:0556 FuEngine             plugin linux_lockdown is not enabled
18:43:47:0556 FuEngine             plugin synaptics_cxaudio is not enabled
18:43:47:0556 FuEngine             plugin acpi_dmar is not enabled
18:43:47:0556 FuEngine             plugin dell_dock is not enabled
18:43:47:0556 FuEngine             plugin cpu is not enabled
18:43:47:0556 FuEngine             Emitting PropertyChanged('Status'='loading')
Loading…                 [-                                      ]18:43:47:0556 FuEngine             using plugins: elantp
18:43:47:0556 FuEngine             USB 0bda:8153 added
18:43:47:0557 FuEngine             failed to find specified plugin dell: no plugin dell found
18:43:47:0558 FuEngine             USB 0bda:0413 added
18:43:47:0558 FuEngine             USB 0bda:0487 added
18:43:47:0559 FuEngine             USB 1d6b:0003 added
18:43:47:0560 FuEngine             USB 1d6b:0002 added
18:43:47:0561 FuEngine             USB 1d6b:0003 added
18:43:47:0562 FuEngine             USB 8087:0029 added
18:43:47:0566 FuEngine             USB 413c:b06e added
18:43:47:0566 FuEngine             failed to find specified plugin dell_dock: no plugin dell_dock found
18:43:47:0566 FuEngine             USB 413c:b06f added
18:43:47:0567 FuEngine             failed to find specified plugin dell_dock: no plugin dell_dock found
18:43:47:0567 FuEngine             USB 0bda:402e added
18:43:47:0570 FuEngine             USB 0bda:5413 added
18:43:47:0572 FuEngine             USB 0bda:5487 added
18:43:47:0573 FuEngine             USB 0c45:6723 added
18:43:47:0576 FuEngine             USB 1050:0407 added
18:43:47:0578 FuEngine             USB 1d6b:0002 added
18:43:47:0580 FuEngine             10 devices with subsystem i2c-adapter
18:43:47:0580 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:02.0/drm/card0/card0-DP-1/i2c-8 added
18:43:47:0581 FuEngine             elantp ignoring: no device file
18:43:47:0581 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:02.0/drm/card0/card0-DP-2/i2c-9 added
18:43:47:0582 FuEngine             elantp ignoring: no device file
18:43:47:0582 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:02.0/drm/card0/card0-eDP-1/i2c-7 added
18:43:47:0582 FuEngine             elantp ignoring: no device file
18:43:47:0583 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:02.0/i2c-3 added
18:43:47:0583 FuEngine             elantp ignoring: no device file
18:43:47:0583 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:02.0/i2c-4 added
18:43:47:0584 FuEngine             elantp ignoring: no device file
18:43:47:0584 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:02.0/i2c-5 added
18:43:47:0585 FuEngine             elantp ignoring: no device file
18:43:47:0585 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:02.0/i2c-6 added
18:43:47:0585 FuEngine             elantp ignoring: no device file
18:43:47:0585 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:15.0/i2c_designware.0/i2c-1 added
18:43:47:0586 FuEngine             elantp ignoring: no device file
18:43:47:0586 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:15.1/i2c_designware.1/i2c-2 added
18:43:47:0586 FuEngine             elantp ignoring: no device file
18:43:47:0587 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:1f.4/i2c-0 added
18:43:47:0587 FuEngine             elantp ignoring: no device file
18:43:47:0588 FuEngine             5 devices with subsystem hidraw
18:43:47:0588 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.0/0003:1050:0407.000A/hidraw/hidraw3 added
18:43:47:0589 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.1/0003:1050:0407.000B/hidraw/hidraw4 added
18:43:47:0590 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:14.0/usb1/1-6/1-6.3/1-6.3.5/1-6.3.5:1.0/0003:413C:B06F.004F/hidraw/hidraw2 added
18:43:47:0592 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:14.0/usb1/1-6/1-6.5/1-6.5:1.0/0003:413C:B06E.004E/hidraw/hidraw1 added
18:43:47:0593 FuEngine             UDEV /sys/devices/pci0000:00/0000:00:15.1/i2c_designware.1/i2c-2/i2c-CUST0001:00/0018:06CB:76AF.0001/hidraw/hidraw0 added
18:43:47:0593 FuEngine             failed to find specified plugin synaptics_rmi: no plugin synaptics_rmi found
18:43:47:0594 FuEngine             Emitting PropertyChanged('Status'='idle')
Loading…                 [***************************************]
No detected devices
```